src/combineReducers.js

```diff
@@ -113,14 +113,15 @@ export default function combineReducers(reducers) {
       throw sanityError
     }
 
+    var hasChanged = false
     if (process.env.NODE_ENV !== 'production') {
       var warningMessage = getUnexpectedStateShapeWarningMessage(state, finalReducers, action)
       if (warningMessage) {
         warning(warningMessage)
+        hasChanged = true
```

Comment:
The issue with this is it changes behavior in development and production. It can cause subtle bugs.